### ISSUE TITLE
Clause match warning

### ISSUE THREAD

**Issue by pikeas** [elixir-sqlite/ecto_sqlite3#157]
After updating to Elixir 1.18:

```
==> ecto_sqlite3
Compiling 4 files (.ex)
      warning: the following clause will never match:

          :text_datetime

      because it attempts to match on the result of:

          :iso8601

      which has type:

          :iso8601

      typing violation found at:
      │
 1392 │         :text_datetime ->
      │         ~~~~~~~~~~~~~~~~~
      │
      └─ lib/ecto/adapters/sqlite3/connection.ex:1392: Ecto.Adapters.SQLite3.Connection.expr/3

Generated ecto_sqlite3 app
```

**Comment by warmwaffles** [comment]
Probably not the only issues to start cropping up because of the new version.

**Comment by warmwaffles** [comment]
Fix shipped with `v0.17.6`
